javagen/src/main/java/com/azure/autorest/template/PatchModelTemplate.java

```diff
@@ -0,0 +1,290 @@
+                        methodBlock.ifBlock(String.format("%s == null", property.getName()), ifBlock ->
+                                ifBlock.line("%s = new HashMap<>();", property.getName()));
+                        methodBlock.line("%s.put(%s, value);", property.getName(), model.getNeedsFlatten() ? "key.replace(\"\\\\.\", \".\")" : "key");
+                    });
+                }
+            }
+        });
+    }
+
+    /**
+     * Adds the property fields to a class.
+     *
+     * @param model      The client model.
+     * @param classBlock The Java class.
+     * @param settings   AutoRest configuration settings.
+     */
+    private static void addProperties(ClientModel model, JavaClass classBlock, JavaSettings settings) {
+        for (ClientModelProperty property : model.getProperties()) {
+
+            classBlock.blockComment(settings.getMaximumJavadocCommentWidth(), (comment) ->
+                    comment.line(property.getDescription()));
+
+//            if (settings.getClientFlattenAnnotationTarget() == JavaSettings.ClientFlattenAnnotationTarget.FIELD && property.getNeedsFlatten()) {
+//                classBlock.annotation("JsonFlatten");
+//            }
+//
+//            // If the property is a polymorphic discriminator for the class add the annotation @JsonTypeId.
+//            // This will indicate to Jackson that the discriminator serialization is determined by the property
+//            // instead of the class level @JsonTypeName annotation. This prevents the discriminator property from
+//            // being serialized twice, once for the class level annotation and again for the property annotation.
+//            if (property.isPolymorphicDiscriminator()) {
+//                classBlock.annotation("JsonTypeId");
+//            }
+
+            if (property.getHeaderCollectionPrefix() != null && !property.getHeaderCollectionPrefix().isEmpty()) {
+                classBlock.annotation("HeaderCollection(\"" + property.getHeaderCollectionPrefix() + "\")");
+            } else if (property.getAnnotationArguments() != null && !property.getAnnotationArguments().isEmpty()) {
+                classBlock.annotation(String.format("JsonProperty(%1$s)", property.getAnnotationArguments()));
+            }
+
+            if (!property.isAdditionalProperties() && property.getClientType() instanceof MapType && settings.isFluent()) {
+                classBlock.annotation("JsonInclude(value = JsonInclude.Include.NON_NULL, content = JsonInclude.Include.ALWAYS)");
+            }
+
+            // handle x-ms-client-default
+            if (property.getDefaultValue() != null) {
+                classBlock.privateMemberVariable(String.format("%1$s %2$s = Option.of(%3$s)", new GenericType("com.azure.core.implementation", "Option", property.getWireType()), property.getName(), property.getDefaultValue()));
```

Comment:
Why should we have default value for a PATCH? The default should be always `null`?

Reply:
Agree, default value should not be used in patch model, I will remove this.